app/src/ui/diff/diff-syntax-mode.ts

```diff
@@ -150,27 +168,38 @@ export class DiffSyntaxMode {
     // The first character of a line in a diff is always going to
     // be the diff line marker so we always take care of that first.
     if (stream.sol()) {
-      const index = stream.next()
+      const tokenKey = stream.next()
 
       if (stream.eol()) {
         state.diffLineIndex++
       }
 
-      if (index === null) {
+      if (tokenKey === null) {
         return null
       }
 
-      const token = TokenNames[index] ?? null
+      const token = TokenNames[tokenKey] ?? null
 
       if (token === null) {
         return null
       }
 
-      let result = getBaseDiffLineStyle(token)
+      const nextLine = stream.lookAhead(1)
+      const nextLineToken =
+        typeof nextLine === 'string' ? TokenNames[nextLine[0]] : null
```

Comment:
If you inspect the type of `nextLineToken` here TypeScript will tell you it's `string | null` which is a lie since an empty line will produce `undefined` (`""[0] === undefined`) and so will a line that starts with a character that doesn't exist in the TokenNames object.

Either we change the type to be `string | undefined` by means of changing the `: null` to `: undefined`

```ts
typeof nextLine === 'string' ? TokenNames[nextLine[0]] : undefined
```

or we keep the current type by converting undefined to null

```ts
typeof nextLine === 'string' ? TokenNames[nextLine[0]] ?? null : null
```

Reply:
Went with everything could be undefined.